src/util/simplify_expr.cpp

```diff
@@ -1777,9 +1777,26 @@ bool simplify_exprt::simplify_byte_extract(byte_extract_exprt &expr)
   const auto bits=
     expr2bits(expr.op(), expr.id()==ID_byte_extract_little_endian);
 
-  // exact match of length only - otherwise we might lose bits of
-  // flexible array members at the end of a struct
-  if(bits.has_value() && mp_integer(bits->size()) == *el_size + *offset * 8)
+  // make sure we don't lose bits with flexible struct members
+  const bool has_flexible_struct_member = has_subtype(
```

Comment:
Nitpick: these appear to be called flexible array members elsewhere, so we might call this `struct_has_flexible_array_member`

Reply:
Thanks, done!